action.php

```diff
@@ -144,9 +145,39 @@ public function handleHtmlEditformOutput($event, $param) {
         $origtext = io_readWikiPage($file,$id);
 
         // Insert original page on the side
-        $form = & $event->data;
-        $pos = $form->findElementByType('wikitext');
-        if ($pos===false) return;
+        $form = $event->data;
+        if (is_a($form, \dokuwiki\Form\Form::class)) {
+            $pos = $form->findPositionByType('textarea'); // new
+            if ($pos===false) return;
+            $this->handleHtmlEditformOutputNG($form, $pos, $origtext);
+        } else {
+            $pos = $form->findElementByType('wikitext'); // old
+            if ($pos===false) return;
+            $this->handleHtmlEditformOutputLegacy($form, $pos, $origtext);
+        }
+    }
+
+    protected function handleHtmlEditformOutputNG($form, $pos, $origtext) {
+        // Before the wikitext...
+        $form->addElement(new dokuwiki\Form\TagOpenElement('div', array('id'=>'wrapper__wikitext','class'=>'hor')), $pos++);
+        // After the wikitext...
+        $pos++;
+        $form->addElement(new dokuwiki\Form\TagCloseElement('div'), $pos++);
+        $form->addElement(new dokuwiki\Form\TagOpenElement('div', array('id'=>'wrapper__sourcetext','class'=>'hor')), $pos++);
+        $origelem = '<textarea id="translate__sourcetext" '.
+                    //buildAttributes($attrs,true).
+                    'class="edit" readonly="readonly" cols="80" rows="10"'.
+                    'style="width:100%;"'.//  height:600px; overflow:auto
```

Comment:
What about height and overflow? Something you tried and you're not sure they're needed?

I got this when I click "edit" without logging in:

![image](https://user-images.githubusercontent.com/273886/162702368-2b046742-2b63-48a0-9638-efc0feba6a19.png)

Reply:
OK, that's fine.

Too much CSS can mess up user stylesheets too, so better leave it as it is.

I just saw "//  height:600px; overflow:auto" commented out, and "//buildAttributes($attrs,true)" too. That's why I mentioned overflow. Perhaps you can delete those if they're not used.

Reply:
> Perhaps you can delete those if they're not used.

he he, that's easily done. Anyway, the last commit (d238a5d61e4271fb721ca00eeaec74a41d9f6c23) made this disappear

Reply:
I think I've resolved the problem. This is no CSS problem, but JS. your JS which is supposed to take care of this is outdated. On current DW, it crashes.

CSS is incomprehensible black magic to me, JS is only a strange but still manageable language. I'll propose a solution for this in another PR.

<img width="1044" alt="image" src="https://user-images.githubusercontent.com/1008292/163101079-e088a79c-8c7d-4634-8786-d39ec009bf69.png">

This was reported as #15.